python/lsst/ip/isr/isrTask.py

```diff
@@ -2097,82 +2207,43 @@ def updateVariance(self, ampExposure, amp, overscanImage=None, ptcDataset=None):
             Exposure to process.
         amp : `lsst.afw.cameraGeom.Amplifier` or `FakeAmp`
             Amplifier detector data.
-        overscanImage : `lsst.afw.image.MaskedImage`, optional.
-            Image of overscan, required only for empirical read noise.
-        ptcDataset : `lsst.ip.isr.PhotonTransferCurveDataset`, optional
-            PTC dataset containing the gains and read noise.
-
-        Raises
-        ------
-        RuntimeError
-            Raised if either ``usePtcGains`` of ``usePtcReadNoise``
-            are ``True``, but ptcDataset is not provided.
-
-            Raised if ```doEmpiricalReadNoise`` is ``True`` but
-            ``overscanImage`` is ``None``.
+        ptcDataset : `lsst.ip.isr.PhotonTransferCurveDataset`
+            Effective PTC dataset containing the gains and read noise.
 
         See also
         --------
         lsst.ip.isr.isrFunctions.updateVariance
         """
-        maskPlanes = [self.config.saturatedMaskName, self.config.suspectMaskName]
-        if self.config.usePtcGains:
-            if ptcDataset is None:
-                raise RuntimeError("No ptcDataset provided to use PTC gains.")
-            else:
-                gain = ptcDataset.gain[amp.getName()]
-                self.log.info("Using gain from Photon Transfer Curve.")
-        else:
-            gain = amp.getGain()
+        ampName = amp.getName()
 
+        # At this point, the effective PTC should have
+        # gain and noise values.
+        gain = ptcDataset.gain[ampName]
         if math.isnan(gain):
```

Comment:
Can this be moved to the effectivePtc method?  It's here because this was historically the first use, but keeping all of the logic in one place is simpler.